packages/vats/src/virtual-purse.js

```diff
@@ -3,12 +3,13 @@ import { E } from '@endo/far';
 import { isPromise } from '@endo/promise-kit';
 
 import {
+  AmountPatternShape,
   AmountShape,
   BrandShape,
   DepositFacetShape,
   NotifierShape,
   PaymentShape,
-} from '@agoric/ertp/src/typeGuards.js';
+} from '@agoric/ertp';
```

Comment:
Truncating the import from `@agoric/ertp/src/typeGuards.js` to `@agoric/ertp` is mostly a drive-by. It is related to this PR only because I needed to add another import name.

Reply:
looks like the vat-bank bundle already includes all of ERTP, so I guess this doesn't matter.

But I did a quick experiment with

```js
import { BrandShape } from '@agoric/ertp/src/typeGuards.js';
console.log(BrandShape);
```

vs importing the whole package:

```
1.0M bundles/bundle-ertp-fun-all.js
596K bundles/bundle-ertp-fun.js
```

`typeGuards.js` only depends on patterns (and hence pass-style, ...) but it doesn't depend on @agoric/store, notifier, nor zones.

Reply:
Attn @kriskowal is there any way our bundler can do better tree shaking on its own without us needing to write more complex imports?

Reply:
Given our constraints, and even in general cases, I believe that writing and using narrowly-scoped modules is the best technique for keeping applications small. Automatic tree-shaking is too automatic. Anything better than what we have requires transforming sources.

For this reason, I discourage using tools like “lodash” or “underscore”. Even Q was a delicate balance where every additional method had to be very likely to be used by any application that used Q.

There is some wiggle room. The constraint is not that every import site must use every exported facility. The constraint is rather that a bundle should overall make use of all of the functionality it contains. So, “pass-style”, while large and indivisible, is okay to the extent that every application that uses “pass-style” is likely to benefit from most of it.

Though, I would like us to become more deliberate about the modules exported at the top-level of every package. Importing through `src` suggests (but does not indicate!) that we did not carefully consider these trade-offs in the design of the public API of the package.